tests/views/test.sign-in.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import TestUtils from 'react/lib/ReactTestUtils';
+
+import { initialUserState } from 'reducers/user';
+import SignIn from 'views/shared/sign-in';
+
+
+describe('SignIn', function() {
+
+  var fakeAccessToken = 'some-oauth-access-token';
+  var fakeTokenSignIn;
+  var fakeUser;
+  var fakeUserSignIn;
+
+  beforeEach(function() {
+    fakeTokenSignIn = sinon.spy();
+    fakeUser = Object.assign({}, initialUserState);
+    fakeUserSignIn = sinon.spy();
+  });
+
+  function mountView({accessToken=fakeAccessToken, user=fakeUser} = {}) {
+    return TestUtils.renderIntoDocument(
+      <SignIn
+        accessToken={accessToken}
+        tokenSignIn={fakeTokenSignIn}
+        user={user}
+        userSignIn={fakeUserSignIn}
+      />
+    );
+  }
+
+  it('should sign in on mount', function() {
+    mountView();
+    assert.equal(fakeTokenSignIn.firstCall.args[0], fakeAccessToken);
+  });
+
+  it('should do a token sign-in with access token', function() {
+    var token = 'some-fxa-token';
+    mountView({accessToken: token});
+    assert.equal(fakeTokenSignIn.firstCall.args[0], token);
+  });
+
+  it('should do a user sign-in without access token', function() {
+    mountView({accessToken: null});
```

Comment:
You may know this but I just learned something cool here. If I were to do `mountView({accessToken: undefined})` then the value of `accessToken` is set to the default, `fakeAccessToken`. That's nice. It's not what I wanted for this case but I actually didn't know that was possible. You can't do that in python; passing `None` would always set it to `None`.